add toggle task definition (#209)
kudobuilder/www: content/docs/developing-operators/tasks.md

**Note:** deleting non-existing resources is always successful. As of version 0.9.0, KUDO will not wait for the resource to be actually removed and will finish the task when the API server accepts the deletion request. So in case of Pods, Kubernetes imposes a default graceful termination period of 30 seconds, however, a delete-task will be done before that. KUDO 0.9.0 will not wait for [resource finalizers](https://kubernetes.io/docs/concepts/workloads/controllers/garbage-collection/) should they exist.

> ## Toggle-Task

Reviewer: Could you please mention Toggle-Tasks [in this paragraph](https://github.com/kudobuilder/kudo.dev/pull/209/files#diff-f9e3827927df0720222e93b2a95594b3R5)?

Author: added to the overview

> Developing featrues for operators sometimes requires enabling and disabling resources.

Reviewer 2: ```suggestion
Developing features for operators sometimes requires enabling and disabling resources.
```

> A toggle-task runs an apply-task or delete-task based on a parameter boolean value.

Reviewer: thought: should we treat the fact that actually apply/delete tasks are used on the inside as an implementation detail? 🤔 
```suggestion
A toggle-task applies or deletes resources based on a parameter boolean value.
```

Let's add a service for an operator based on the parameter `enable-service`:

```yaml
tasks:
  - name: app-service
    kind: Toggle
    spec:
      parameter: enable-service
      resources:
        - service.yaml
```

> Based on the `enable-service` value the task will either `apply` or `delete` the resources defined in `templates/service.yaml`

Reviewer: ```suggestion
This task will either apply or delete the resources defined in `templates/service.yaml` based on the `enable-service` parameter value.
```

If the `enable-service` parameter evaluates to `true` the task named `app-service` will create a service resource defined in `templates/service.yaml`. In case the `enable-service` parameter evaluates to `false`, the task named `app-service` will delete the service resource defined in `templates/service.yaml`.

> 

Reviewer 2: It would be useful to note that the parameter must be defined in the params.yaml file.

Reviewer: ...and mention what happens if the parameter type is not boolean.

Author: added the param requirement
